src/Service/QuapService.php

```diff
@@ -0,0 +1,163 @@
+     * @param EntityManagerInterface $em
+     */
+    public function __construct(
+        QuestionnaireRepository $questionnaireRepository,
+        AspectRepository $aspectRepository,
+        QuestionRepository $questionRepository,
+        HelpRepository $helpRepository,
+        LinkRepository $linkRepository,
+        WidgetQuapRepository $quapRepository,
+        EntityManagerInterface $em
+    ) {
+        $this->questionnaireRepository = $questionnaireRepository;
+        $this->aspectRepository = $aspectRepository;
+        $this->questionRepository = $questionRepository;
+        $this->helpRepository = $helpRepository;
+        $this->linkRepository = $linkRepository;
+        $this->quapRepository = $quapRepository;
+        $this->em = $em;
+    }
+
+    /**
+     * @param string $type
+     * @param string $locale
+     * @param string $dateTime
+     * @return Questionnaire|null
+     */
+    public function getQuestionnaireByType(string $type, string $locale, string $dateTime): ?Questionnaire
+    {
+        $questionnaire = $this->questionnaireRepository->findOneBy(["type" => $type]);
+        $questionnaire->setAspects(new ArrayCollection($this->aspectRepository->getExisting($questionnaire->getId(), $dateTime)));
+
+        /** @var Aspect $aspect */
+        foreach ($questionnaire->getAspects() as $aspect) {
+            $aspect->setQuestions(new ArrayCollection($this->questionRepository->getExisting($aspect->getId(), $dateTime)));
+
+            /** @var Question $question */
+            foreach ($aspect->getQuestions() as $question) {
+                $question->setHelp(new ArrayCollection($this->helpRepository->getExisting($question->getId(), $dateTime)));
+
+                /** @var Help $help */
+                foreach ($question->getHelp() as $help) {
+                    switch ($locale) {
+                        case (str_contains($locale, "it")):
+                            $help->setLinksIt(new ArrayCollection($this->linkRepository->findBy([ 'helpIt' => $help])));
+                            break;
+                        case (str_contains($locale, "fr")):
+                            $help->setLinksFr(new ArrayCollection($this->linkRepository->findBy([ 'helpFr' => $help])));
+                            break;
+                        default:
+                            $help->setLinksDe(new ArrayCollection($this->linkRepository->findBy([ 'helpDe' => $help])));
+                            break;
+                    }
+                }
+            }
+        }
+
+        return $questionnaire;
+    }
+
+    public function submitAnswers(Group $group, array $json): WidgetQuap
+    {
+        $widgetQuap = $this->quapRepository->findOneBy([
+            "dataPointDate" => null,
+            "group" => $group->getId()
+        ]);
+
+        if (!$widgetQuap) {
+            // TODO get correct questionnaire
+            $questionnaire = $this->questionnaireRepository->find(1);
```

Comment:
Is this a pending TODO or will this be implemented later? If there's something missing please add it or elaborate on the comment.

Reply:
This cannot be implemented yet but will be implemented asap